src/bpftrace.cpp

```diff
@@ -48,53 +48,67 @@ namespace bpftrace {
 DebugLevel bt_debug = DebugLevel::kNone;
 bool bt_verbose = false;
 volatile sig_atomic_t BPFtrace::exitsig_recv = false;
+const int FMT_BUF_SZ = 512;
 
-int format(char * s, size_t n, const char * fmt, std::vector<std::unique_ptr<IPrintable>> &args) {
+std::string format(std::string fmt,
+                   std::vector<std::unique_ptr<IPrintable>> &args)
+{
+  std::string retstr;
+  int n = FMT_BUF_SZ;
```

Comment:
I prefer avoiding ambiguously named temporaries like `n`. It's ok as a loop counter but usually not as a function-scope variable. It forces the reader to keep track of unnecessary details.

I think using `buffer.capacity()` would be better.